### ISSUE TITLE
mvn clean package fails in StreamCommandTests.java behind corporate firewall

### ISSUE THREAD

**Issue by ssherwood** [spring-cloud/spring-cloud-dataflow#103]
`mvn clean package` fails the test in StreamCommandTests.java on line 36:

```
-------------------------------------------------------------------------------
Test set: org.springframework.cloud.dataflow.shell.command.StreamCommandTests
-------------------------------------------------------------------------------
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 23.223 sec <<< FAILURE! - in org.springframework.cloud.dataflow.shell.command.StreamCommandTests
testStreamLifecycleForTickTock(org.springframework.cloud.dataflow.shell.command.StreamCommandTests)  Time elapsed: 20.437 sec  <<< ERROR!
java.lang.IllegalArgumentException: Failure.  CommandResult = CommandResult [success=false, result=null, exception=org.springframework.cloud.dataflow.rest.client.DataFlowClientException: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact org.springframework.cloud.stream.module:log-sink:jar:exec:1.0.0.BUILD-SNAPSHOT from/to repository 1 (https://repo.spring.io/libs-snapshot): connect timed out
]
	at org.springframework.util.Assert.isTrue(Assert.java:68)
	at org.springframework.cloud.dataflow.shell.AbstractShellIntegrationTest$DataFlowShell.executeCommand(AbstractShellIntegrationTest.java:195)
	at org.springframework.cloud.dataflow.shell.command.StreamCommandTemplate.doCreate(StreamCommandTemplate.java:86)
	at org.springframework.cloud.dataflow.shell.command.StreamCommandTemplate.create(StreamCommandTemplate.java:66)
	at org.springframework.cloud.dataflow.shell.command.StreamCommandTests.testStreamLifecycleForTickTock(StreamCommandTests.java:36)
```

It appears to be attempting to pull an artifact from `https://repo.spring.io/libs-snapshot` which I believe is failing due to fact that I am required to operate behind a corporate firewall.  Internally we have an artifact repository that acts as a proxy in these cases and have maven configured via settings.xml.

**Comment by davidturing** [comment]
Hit the same problem.
It seems spring-cloud-dataflow is still in dev and not suitable for production yet (Production usually do not allow to connect to internet).

**Comment by sabbyanandan** [comment]
Hi @ssherwood and @davidturing:

The `master` is always setup to pull the `SNAPSHOT` dependencies, so we could try few things here.

* Perhaps you could change the relevant artifacts to respective milestone releases, so you wouldn't have to rely on the snapshots from `https://repo.spring.io/libs-snapshot`
* You could always skip the test runs when running behind the firewall with maven target `-DskipTests`

With that said, we do realize there could be a problem with the use of `AetherModuleResolver` to resolve and download the module jar's at runtime. We will be investigating this issue via: spring-cloud/spring-cloud-stream#272

Please share your findings.

**Comment by ilayaperumalg** [comment]
Hi,

Please note that the artifact download behind the proxy using the `out-of-process` deployer (which is the default local deployer) is still an issue based on the conversation here: https://github.com/spring-cloud/spring-cloud-dataflow/issues/420

I believe the `out-of-process` deployer is one that is used in the above test run. We will investigate the issue: https://github.com/spring-cloud/spring-cloud-dataflow/issues/420

**Comment by sabbyanandan** [comment]
Proxy settings is addressed via spring-cloud/spring-cloud-dataflow#439 and documentation is [here](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc#deploying-spring-cloud-data-flow) for reference. 

@ssherwood and @davidturing: if you're still facing issues with it, please open a new issue.
